ps_mbo.php

```diff
@@ -95,169 +81,184 @@ public function __construct() {
+                'controller' => Tools::getValue('controller')
+            ));
+
+            return $this->context->smarty->fetch($this->template_dir . '/toolbar.tpl');
+        }
+    }
+
+    private function getAddonsConnectToolbar()
+    {
+        $container = SymfonyContainer::getInstance();
         $addonsProvider = $container->get('prestashop.core.admin.data_provider.addons_interface');
-        $addonsConnect = array();
+        $addonsConnect = [];
+
+        $authenticated = $addonsProvider->isAddonsAuthenticated();
 
-		$authenticated = $addonsProvider->isAddonsAuthenticated();
-		
         if ($addonsProvider->isAddonsAuthenticated()) {
             $addonsEmail = $addonsProvider->getAddonsEmail();
-			return array(
-				'connected' => true,
-				'email' => $addonsEmail['username_addons'],
-				'logout_url' => $container->get('router')->generate('admin_addons_logout', [], UrlGeneratorInterface::ABSOLUTE_URL)
-			);
-        } else {
-			return array(
-				'connected' => false,
-				'login_url' => $container->get('router')->generate('admin_addons_login', [], UrlGeneratorInterface::ABSOLUTE_URL)
-			);
+            return array(
+                'connected' => true,
+                'email' => $addonsEmail['username_addons'],
+                'logout_url' => $container->get('router')->generate('admin_addons_logout', [], UrlGeneratorInterface::ABSOLUTE_URL)
+            );
         }
+
+        return array(
+            'connected' => false,
+            'login_url' => $container->get('router')->generate('admin_addons_login', [], UrlGeneratorInterface::ABSOLUTE_URL)
+        );
     }
-	
-	private function installTab() {
-        $tab = new Tab();
-        $tab->active = 1;
-        $tab->class_name = 'AdminPsMboModule';
-        $tab->name = array();
-        foreach (Language::getLanguages(true) as $lang) {
-            $tab->name[$lang['id_lang']] = $this->displayName;
-        }
-        unset($lang);
-        $tab->id_parent = -1;
-        $tab->module = $this->name;
-        $tab->add();
-		
-        $tab = new Tab();
-        $tab->active = 1;
-        $tab->class_name = 'AdminPsMboTheme';
-        $tab->name = array();
-        foreach (Language::getLanguages(true) as $lang) {
-            $tab->name[$lang['id_lang']] = $this->displayName;
-        }
-        unset($lang);
-        $tab->id_parent = -1;
-        $tab->module = $this->name;
-        $tab->add();
+
+    private function checkTabsPositions()
+    {
+        if (!Configuration::get(self::POSITION_CHECKED)) {
+            $tabs = [
+                'AdminPsMboModule' => 'AdminModulesCatalog',
+                'AdminPsMboTheme' => 'AdminThemesCatalog',
+            ];
+            $updated = false;
+            foreach ($tabs as $new => $old) {
+                $newTabId = Tab::getIdFromClassName($new);
+                if (!empty($newTabId)) {
+                    $oldTabId = Tab::getIdFromClassName($old);
```

Comment:
What do you think about loading the two tabs at the same time before verifying?
You could get rid of one level of indentation.

```php
                $newTabId = Tab::getIdFromClassName($new);
                $oldTabId = Tab::getIdFromClassName($old);
                if (!empty($newTabId) && !empty($oldTabId)) {
```

Reply:
I manage tabs separately to prevent error while migrating (and in case they use the old mbo module). I had some problem when trying to manage both -_-'